app/assets/javascripts/spree/frontend/solidus_paypal_commerce_platform/button_actions.js

```diff
@@ -11,6 +11,56 @@ SolidusPaypalCommercePlatform.postOrder = function(payment_method_id) {
   })
 }
 
+SolidusPaypalCommercePlatform.createAndSendOrder = function(payment_method_id) {
+  return SolidusPaypalCommercePlatform.createOrder().then(function(){
+    return SolidusPaypalCommercePlatform.sendOrder(payment_method_id)
+  })
+}
+
+SolidusPaypalCommercePlatform.createOrder = function() {
+  var data = {
+    order: {
+      line_items_attributes: [{
+        variant_id: SolidusPaypalCommercePlatform.getVariantId(),
+        quantity: SolidusPaypalCommercePlatform.quantity_field.value
+      }]
+    }
+  }
+
+  if(typeof SolidusPaypalCommercePlatform.user_id !== 'undefined'){
```

Comment:
nit: one neat trick with JS is doing `!= null` to check that a value is not `null` nor `undefined` (which is the desired outcome most of the time).

Reply:
Oh that's handy, thanks!